celery/backends/base.py

```diff
@@ -453,10 +454,8 @@ def store_result(self, task_id, result, state,
                             self.max_sleep_between_retries_ms, True) / 1000
                         self._sleep(sleep_amount)
                     else:
-                        reraise(
-                            BackendStoreError,
+                        raise_with_context(
                             BackendStoreError("failed to store result on the backend", task_id=task_id, state=state),
-                            traceback,
```

Comment:
@auvipy I've been looking at this code, and reraise-ing this traceback is not correct.

I think the intention was to include the context of the failed request from the retry - not the failure of the actual task/result.

Reply:
there's already a utility to re-raise an exception from another context in serialization